### PR TITLE
add s3 storage option

### PR DESCRIPTION
**Pull request by hjmodi**

### CODE REVIEW

File: src/main/java/org/commonjava/service/storage/config/StorageServiceConfig.java [hunk 1]
```diff
@@ -35,4 +40,12 @@ public interface StorageServiceConfig
 
     @WithName( "removableFilesystemPattern" )
     String removableFilesystemPattern();
+
+    @WithName( "type" )
+    @WithDefault( "" )
```
Comment (ruhan1) [inline src/main/java/org/commonjava/service/storage/config/StorageServiceConfig.java]: maybe just set the default to nfs, and manually update the config for mp+ , so it won't affect current environments.

Comment (ruhan1) [reply src/main/java/org/commonjava/service/storage/config/StorageServiceConfig.java]: all others look good to me.

Comment (hjmodi) [reply src/main/java/org/commonjava/service/storage/config/StorageServiceConfig.java]: updated

**Review by ruhan1 — APPROVED** [review]
LGTM

**Pull request merged**